backend/decky_loader/browser.py

```diff
@@ -213,7 +213,19 @@ async def _install(self, artifact: str, name: str, version: str, hash: str):
                     return
 
                 else:
-                    name = sub(r"/.+$", "", plugin_json_list[0])
+                    plugin_json_file = plugin_json_list[0]
+                    name = sub(r"/.+$", "", plugin_json_file)
+                    try:
+                        with plugin_zip.open(plugin_json_file) as f:
+                            plugin_json_data = json.loads(f.read().decode('utf-8'))
+                            plugin_name_from_plugin_json = plugin_json_data.get('name')
+                            if plugin_name_from_plugin_json and plugin_name_from_plugin_json.strip():
```

Comment:
Imo either use the stripped output or don't use strip at all. Do we need safeguarding against plugin.json's that have a space as their name?

Reply:
Was meant to be a "null or witespace" or "blank" check.
First part of the condition will exclude missing `"name"` (None) and empty strings (`"name": ""`)
Second part will exclude blank strings (`"name": " "`).
Doesn't hurt I think, but for sure not needed. No problem to remove the stripped check.

Reply:
Was more so asking because ' my plugin ' would get through this check, but unsure if that's a good idea or would cause issues later. Either way also on the plugin dev imo if they enter something invalid in the name field.

Reply:
I understand. Plugin name `" my plugin "` wouldn't cause issues later (at least during this code path) because it is used in `find_plugin_folder` function that only does `if plugin["name"] == name:` and it will find the plugin folder without problems.

Maybe I should only check for None and not even try to deal with empty strings.
`if plugin_name_from_plugin_json is not None:`